### PR TITLE
Add support for search shortcut to all docks

### PR DESCRIPTION
**Pull request by passivestar**
Adds support for Ctrl+F to scene, inspector and node docks, see https://github.com/godotengine/godot-proposals/discussions/8845

https://github.com/godotengine/godot/assets/60579014/19d7b72c-75be-4225-8579-ff5aaf007a9d

*Bugsquad edit:*
- Closes https://github.com/godotengine/godot-proposals/discussions/8845

### CODE REVIEW

**Comment by ryevdokimov** [comment]
Works pretty good.

**Review by Calinou — APPROVED** [review]
Tested locally, it works as expected.

PS: If you have time, I'd appreciate a review on https://github.com/godotengine/godot/pull/81177 which is related to this PR :slightly_smiling_face:

File: editor/connections_dialog.cpp [hunk 1]
```diff
@@ -1543,6 +1557,7 @@ ConnectionsDock::ConnectionsDock() {
 	tree->connect("item_selected", callable_mp(this, &ConnectionsDock::_tree_item_selected));
 	tree->connect("item_activated", callable_mp(this, &ConnectionsDock::_tree_item_activated));
 	tree->connect("gui_input", callable_mp(this, &ConnectionsDock::_rmb_pressed));
+	tree->connect("gui_input", callable_mp(this, &ConnectionsDock::_connections_gui_input));
```
Comment (Calinou) [inline editor/connections_dialog.cpp:1560]: Connecting a single signal to multiple functions looks a bit strange to me. Do we use this pattern frequently elsewhere?

Comment (KoBeWi) [reply editor/connections_dialog.cpp:1560]: `_rmb_pressed` code should be moved to gui input method.

Comment (passivestar) [reply editor/connections_dialog.cpp]: Fixed

File: editor/inspector_dock.cpp [hunk 2]
```diff
@@ -614,6 +614,26 @@ void InspectorDock::apply_script_properties(Object *p_object) {
 	stored_properties.clear();
 }
 
+void InspectorDock::input(const Ref<InputEvent> &p_event) {
```
Comment (YuriSizov) [inline editor/inspector_dock.cpp:617]: This should probably use `shortcut_input` (and the corresponding set processing needs to be called in the constructor).

**Comment by YuriSizov** [comment]
Also needs a rebase because there was another PR touching the input handling in the connections dialog 🙃

File: editor/scene_tree_dock.cpp [hunk 3]
```diff
@@ -162,6 +162,20 @@ void SceneTreeDock::shortcut_input(const Ref<InputEvent> &p_event) {
 	accept_event();
 }
 
+void SceneTreeDock::_scene_tree_gui_input(Ref<InputEvent> p_event) {
+	Ref<InputEventKey> key = p_event;
+
+	if (!key.is_valid() || !key->is_pressed() || key->is_echo()) {
```
Comment (KoBeWi) [inline editor/scene_tree_dock.cpp:168]:
```suggestion
	if (key.is_null() || !key->is_pressed() || key->is_echo()) {
```

File: editor/inspector_dock.cpp [hunk 4]
```diff
@@ -614,6 +614,26 @@ void InspectorDock::apply_script_properties(Object *p_object) {
 	stored_properties.clear();
 }
 
+void InspectorDock::shortcut_input(const Ref<InputEvent> &p_event) {
+	ERR_FAIL_COND(p_event.is_null());
+
+	Ref<InputEventKey> key = p_event;
+
+	if (!key.is_valid() || !key->is_pressed() || key->is_echo()) {
```
Comment (KoBeWi) [inline editor/inspector_dock.cpp:622]:
```suggestion
	if (key.is_null() || !key->is_pressed() || key->is_echo()) {
```

**Review by KoBeWi — APPROVED** [review]
Looks fine.

The connections dialog also does not support Ctrl+F shortcut. Technically it's not a dock though, so it can be added later.
image

File: editor/connections_dialog.cpp [hunk 5]
```diff
@@ -1203,58 +1203,68 @@ void ConnectionsDock::_slot_menu_about_to_popup() {
...
+			if (item && _get_item_type(*item) == TREE_ITEM_TYPE_CONNECTION) {
+				Connection connection = item->get_metadata(0);
+				_disconnect(connection);
+				update_tree();
+
+				// Stop the Delete input from propagating elsewhere.
+				accept_event();
+				return;
+			}
+			// Handle Ctrl+F press.
```
Comment (AThousandShips) [inline editor/connections_dialog.cpp:1221]: This should be moved down, it is now part of the `connections_editor/disconnect` shortcut

File: editor/connections_dialog.cpp [hunk 6]
```diff
@@ -1203,58 +1203,68 @@ void ConnectionsDock::_slot_menu_about_to_popup() {
...
-		TreeItem *item = tree->get_selected();
-		if (item && _get_item_type(*item) == TREE_ITEM_TYPE_CONNECTION) {
-			Connection connection = item->get_metadata(0);
-			_disconnect(connection);
-			update_tree();
+	const Ref<InputEventKey> &key = p_event;
+
+	if (key.is_valid() && key->is_pressed() && !key->is_echo()) {
+		// Handle Delete press.
+		if (ED_IS_SHORTCUT("connections_editor/disconnect", p_event)) {
```
Comment (AThousandShips) [inline editor/connections_dialog.cpp:1210]:
```suggestion
		if (ED_IS_SHORTCUT("connections_editor/disconnect", p_event)) { // Handle Delete press.
```

File: editor/connections_dialog.cpp [hunk 7]
```diff
@@ -1203,58 +1203,68 @@ void ConnectionsDock::_slot_menu_about_to_popup() {
...
+				Connection connection = item->get_metadata(0);
+				_disconnect(connection);
+				update_tree();
+
+				// Stop the Delete input from propagating elsewhere.
+				accept_event();
+				return;
+			}
+			// Handle Ctrl+F press.
+		} else if (ED_IS_SHORTCUT("editor/open_search", p_event)) {
```
Comment (AThousandShips) [inline editor/connections_dialog.cpp:1222]:
```suggestion
		} else if (ED_IS_SHORTCUT("editor/open_search", p_event)) { // Handle Ctrl+F press.
```

File: editor/connections_dialog.cpp (hunk 6 above)
Comment (YuriSizov) [reply editor/connections_dialog.cpp:1210]: Honestly, I would probably get rid of both comment lines you've highlighted. I'm not sure they add anything. It's already clear what we're handling from the code line itself.

Comment (AThousandShips) [reply editor/connections_dialog.cpp:1210]: I'd agree

**Review by YuriSizov — APPROVED** [review]
If you address the comments above, it should be good to go!

File: editor/connections_dialog.cpp (hunk 6 above)
Comment (passivestar) [reply editor/connections_dialog.cpp]: Ah, I already pushed these suggestions. Should I delete them?

**Review by AThousandShips — APPROVED** [review]

**Comment by passivestar** [comment]
> Thanks, and congrats on your first merged Godot contribution! Small but handy usability improvement 😎

Thanks! <3

**Comment by solarite1797** [comment]
ISNT THIS THE pull request where you added blender into godot

**Comment by keetsta** [comment]
It could be

**Pull request merged**